### PR TITLE
Create a noisy environment tester (connect and disconnect each 1 second)

### PR DESCRIPTION
**Pull request by eslam-cloud2nd**
Create a noisy environment tester for this Issue [48395](https://github.com/zephyrproject-rtos/zephyr/pull/48395)
The tester's purpose is to make the peripheral notify the central with a high rate of data and, at the same time, force the central to disconnect and reconnect to the peripheral each second.
Also, add two log messages to track `tx_meta_data_alloc` and `tx_meta_data_free` allocation count.

Signed-off-by: Eslam Essa [eslam@cloud2gnd.com](mailto:eslam@cloud2gnd.com)

### CODE REVIEW

File: tests/bluetooth/bsim_bt/bism_test_noisy_env/script/noisy_env.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+peripheral_app_name="peripheral_hr"
+bsim_central_exe_name="bs_nrf52_bsim_bluetooth_central"
+bsim_peripheral_exe_name="bs_nrf52_bsim_bluetooth_peripheral"
+cd ${testing_apps_loc}
+if [ ! -d "${central_app_name}" -o ! -d "${peripheral_app_name}" ]; then
+    echo -e "ERR! \e[91mOne or more test applications couldn't be found\e[39m"
+    exit 1
+fi
+#Remove old builds if they exist
+find . -type d -name 'build' -exec rm -rf {} +
```
Comment (aescolar) [inline tests/bluetooth/bsim_bt/bism_test_noisy_env/script/noisy_env.sh:31]: 1. Please use the compile.sh script to build your app (you can see there other examples)
you should not be building and (less) deleting folder or build results from the script that runs the test 
2. you need to sign off the commit, add a comment message body, and correct all the checkpatch violations

**Review by jhedberg — DISMISSED** [review]

File: tests/bluetooth/bsim_bt/bism_test_noisy_env/script/noisy_env.sh (hunk 1 above)
Comment (eslam-cloud2nd) [reply tests/bluetooth/bsim_bt/bism_test_noisy_env/script/noisy_env.sh:31]: > 1. Please use the compile.sh script to build your app (you can see there other examples)
>    you should not be building and (less) deleting folders or build results from the script that runs the test
> 2. you need to sign off the commit, add a commit message body, and correct all the checkpatch violations

1- Done; new script added for compiling `_compile.sh` and `run_noisy_env.sh` for running the tester
2- Done

**Comment by aescolar** [comment]
Hi @eslam-cloud2nd, let me ask a relatively fundamental question before going further:
what is the drive for adding this new test upstream and have it run in CI?

**Comment by eslam-cloud2nd** [comment]
> Hi @eslam-cloud2nd, let me ask a relatively fundamental question before going further: what is the drive for adding this new test upstream and have it run in CI?

The purpose of this tester is to make sure that the issue is solved & to include it with system testers.
No, it didn't run in CI

**Review by aescolar — CHANGES_REQUESTED** [review]
I'm not really sure if you intend for these set of PRs to be merged (#53396 #53136 #53134 #52933 #52927), but to avoid confusion (as others are reviewing them also) I will write a common review here that applies to all of them:
These changes are not mergable as they are. The fundamental issues are:
* These new "tests" are not real tests, they don't pass or fail, they seem to be samples meant to trigger some issue for reproduction. The value of having that upstream seems dubious.
* All test are misplacing their files (samples in test folder, tests not explicitly meant for babblesim in the tests/bluetooth/bsim_bt/ folder)
* Most (or all) tests are adding their own test scripts which would be run in CI, but they all are doing things they should not and which we would not want done in CI or done in that way. There is plenty of extraneous files and things that  seem to be there by mistake (including a zephyr.exe).

If the intention with this PRs was just to share some branches with somebody else so they could reproduce some issue, note that you can just create your own branches in your github fork and share those, you don't need to send PRs upstream for review.
If that is the case, please close this PRs to avoid confusion.

**Pull request closed without merging**